### ISSUE TITLE
linting error while using ../chipper/eslint/sim_es6_eslintrc_review.js"

### ISSUE THREAD

**Issue by veillette** [phetsims/gas-properties#114]
While performing the code-review (#109), we noticed that although it passes the lint test using sim_es6_eslintrc.js, it fails the lint test while using sim_es6_eslintrc_review.js.

```
Running "lint" task

C:\Users\veillette\phetsims\gas-properties\js\common\model\BaseContainer.js
  164:1  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\common\model\CollisionCounter.js
  122:1  error  Line 122 exceeds the maximum line length of 120  max-len

C:\Users\veillette\phetsims\gas-properties\js\common\model\CollisionDetector.js
   67:22  error  'createRegions' was used before it was defined                  no-use-before-define
  110:7   error  'assignParticlesToRegions' was used before it was defined       no-use-before-define
  115:11  error  'doParticleParticleCollisions' was used before it was defined   no-use-before-define
  138:1   error  Line 138 exceeds the maximum line length of 120                 max-len
  138:11  error  'doParticleContainerCollisions' was used before it was defined  no-use-before-define
  243:11  error  'adjustParticleLocation' was used before it was defined         no-use-before-define
  245:11  error  'adjustParticleLocation' was used before it was defined         no-use-before-define
  266:11  error  'adjustParticleSpeed' was used before it was defined            no-use-before-define
  267:11  error  'adjustParticleSpeed' was used before it was defined            no-use-before-define

C:\Users\veillette\phetsims\gas-properties\js\common\model\GasPropertiesContainer.js
   49:1  error  Line 49 exceeds the maximum line length of 120   max-len
  173:1  error  Line 173 exceeds the maximum line length of 120  max-len
  188:1  error  Trailing spaces not allowed                      no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\common\model\GasPropertiesModel.js
  178:1  error  Line 178 exceeds the maximum line length of 120  max-len
  360:1  error  Trailing spaces not allowed                      no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\common\model\ParticleSystem.js
  41:78  error  Trailing spaces not allowed                     no-trailing-spaces
  43:66  error  Trailing spaces not allowed                     no-trailing-spaces
  62:1   error  Line 62 exceeds the maximum line length of 120  max-len
  68:1   error  Line 68 exceeds the maximum line length of 120  max-len

C:\Users\veillette\phetsims\gas-properties\js\common\model\PressureGauge.js
  146:1  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\common\model\PressureModel.js
  39:1  error  Line 39 exceeds the maximum line length of 120  max-len
  89:1  error  Line 89 exceeds the maximum line length of 120  max-len

C:\Users\veillette\phetsims\gas-properties\js\common\view\CollisionCounterCheckbox.js
  48:5  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\common\view\ContainerResizeDragListener.js
  39:1  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\common\view\DimensionalArrowsNode.js
  101:5  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\common\view\EraseParticlesButton.js
  48:1  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\common\view\GasPropertiesBicyclePumpNode.js
  47:1  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\common\view\GasPropertiesIconFactory.js
  325:14  error  'createParticleIcon' was used before it was defined   no-use-before-define
  338:14  error  'createParticleIcon' was used before it was defined   no-use-before-define
  351:14  error  'createParticleIcon' was used before it was defined   no-use-before-define
  364:14  error  'createParticleIcon' was used before it was defined   no-use-before-define
  373:14  error  'createToolIcon' was used before it was defined       no-use-before-define
  382:14  error  'createToolIcon' was used before it was defined       no-use-before-define
  398:11  error  'createParticleIcon' was used before it was defined   no-use-before-define
  399:11  error  'createHistogramIcon' was used before it was defined  no-use-before-define

C:\Users\veillette\phetsims\gas-properties\js\common\view\GasPropertiesParticlesNode.js
  43:1  error  Line 43 exceeds the maximum line length of 120  max-len

C:\Users\veillette\phetsims\gas-properties\js\common\view\GasPropertiesScreenView.js
  119:13  error  'particlesNode' was used before it was defined   no-use-before-define
  131:13  error  'particlesNode' was used before it was defined   no-use-before-define
  178:1   error  Line 178 exceeds the maximum line length of 120  max-len
  184:1   error  Line 184 exceeds the maximum line length of 120  max-len
  201:1   error  Line 201 exceeds the maximum line length of 120  max-len

C:\Users\veillette\phetsims\gas-properties\js\common\view\ParticlesAccordionBox.js
  97:1  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\common\view\ParticlesNode.js
  82:9  error  'drawParticles' was used before it was defined  no-use-before-define

C:\Users\veillette\phetsims\gas-properties\js\common\view\ParticleTypeRadioButtonGroup.js
  59:1  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\common\view\StopwatchCheckbox.js
  48:5  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\common\view\StopwatchNode.js
  61:1  error  Trailing spaces not allowed  no-trailing-spaces
  89:1  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\diffusion\model\DiffusionCollisionDetector.js
  55:1  error  Line 55 exceeds the maximum line length of 120  max-len
  57:1  error  Line 57 exceeds the maximum line length of 120  max-len

C:\Users\veillette\phetsims\gas-properties\js\diffusion\model\DiffusionModel.js
  117:11  error  'updateMassAndTemperature' was used before it was defined  no-use-before-define
  122:11  error  'updateMassAndTemperature' was used before it was defined  no-use-before-define
  136:9   error  'updateRadius' was used before it was defined              no-use-before-define
  139:9   error  'updateRadius' was used before it was defined              no-use-before-define
  226:11  error  'addParticles' was used before it was defined              no-use-before-define
  284:1   error  Line 284 exceeds the maximum line length of 120            max-len
  285:1   error  Line 285 exceeds the maximum line length of 120            max-len

C:\Users\veillette\phetsims\gas-properties\js\diffusion\model\ParticleFlowRate.js
  36:18  error  Trailing spaces not allowed                     no-trailing-spaces
  94:1   error  Line 94 exceeds the maximum line length of 120  max-len

C:\Users\veillette\phetsims\gas-properties\js\diffusion\view\DiffusionSettingsNode.js
   49:1   error  Line 49 exceeds the maximum line length of 120    max-len
   65:44  error  'QuantityControl' was used before it was defined  no-use-before-define
   75:31  error  'QuantityControl' was used before it was defined  no-use-before-define
   86:33  error  'QuantityControl' was used before it was defined  no-use-before-define
   96:45  error  'QuantityControl' was used before it was defined  no-use-before-define
  136:1   error  Line 136 exceeds the maximum line length of 120   max-len
  165:1   error  Line 165 exceeds the maximum line length of 120   max-len
  166:1   error  Line 166 exceeds the maximum line length of 120   max-len

C:\Users\veillette\phetsims\gas-properties\js\diffusion\view\DividerNode.js
  53:1  error  Trailing spaces not allowed  no-trailing-spaces
  55:1  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\diffusion\view\ParticleFlowRateNode.js
  49:53  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\energy\model\AverageSpeedModel.js
   33:1   error  Line 33 exceeds the maximum line length of 120    max-len
  113:36  error  'getAverageSpeed' was used before it was defined  no-use-before-define
  114:36  error  'getAverageSpeed' was used before it was defined  no-use-before-define

C:\Users\veillette\phetsims\gas-properties\js\energy\model\HistogramsModel.js
   29:1   error  Line 29 exceeds the maximum line length of 120           max-len
  148:1   error  Trailing spaces not allowed                              no-trailing-spaces
  150:36  error  'getSpeedValues' was used before it was defined          no-use-before-define
  151:36  error  'getSpeedValues' was used before it was defined          no-use-before-define
  154:44  error  'getKineticEnergyValues' was used before it was defined  no-use-before-define
  155:44  error  'getKineticEnergyValues' was used before it was defined  no-use-before-define
  156:1   error  Trailing spaces not allowed                              no-trailing-spaces
  170:9   error  'samplesToBinCounts' was used before it was defined      no-use-before-define
  172:9   error  'samplesToBinCounts' was used before it was defined      no-use-before-define
  174:9   error  'sumBinCounts' was used before it was defined            no-use-before-define
  178:9   error  'samplesToBinCounts' was used before it was defined      no-use-before-define
  180:9   error  'samplesToBinCounts' was used before it was defined      no-use-before-define
  182:9   error  'sumBinCounts' was used before it was defined            no-use-before-define
  243:1   error  Line 243 exceeds the maximum line length of 120          max-len

C:\Users\veillette\phetsims\gas-properties\js\energy\view\AverageSpeedAccordionBox.js
  88:22  error  Trailing spaces not allowed                     no-trailing-spaces
  93:1   error  Line 93 exceeds the maximum line length of 120  max-len
  94:1   error  Line 94 exceeds the maximum line length of 120  max-len

C:\Users\veillette\phetsims\gas-properties\js\energy\view\EnergyAccordionBox.js
  36:1  error  Line 36 exceeds the maximum line length of 120  max-len

C:\Users\veillette\phetsims\gas-properties\js\energy\view\EnergyScreenView.js
  49:59  error  Trailing spaces not allowed                     no-trailing-spaces
  62:1   error  Line 62 exceeds the maximum line length of 120  max-len

C:\Users\veillette\phetsims\gas-properties\js\energy\view\EnergyToolsPanel.js
  56:1  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\energy\view\HistogramNode.js
   30:1   error  Trailing spaces not allowed  no-trailing-spaces
  132:22  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\energy\view\InjectionTemperatureAccordionBox.js
  104:25  error  'temperatureLayoutFunction' was used before it was defined  no-use-before-define
  150:5   error  Trailing spaces not allowed                                 no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\energy\view\KineticEnergyAccordionBox.js
  30:1  error  Line 30 exceeds the maximum line length of 120  max-len

C:\Users\veillette\phetsims\gas-properties\js\energy\view\SpeedAccordionBox.js
  30:1  error  Line 30 exceeds the maximum line length of 120  max-len

C:\Users\veillette\phetsims\gas-properties\js\explore\view\ExploreToolsPanel.js
  61:1  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\ideal\view\AnimatedHeaterCoolerNode.js
  113:60  error  A function with a name starting with an uppercase letter should only be used as a constructor  new-cap
  114:1   error  Line 114 exceeds the maximum line length of 120                                                max-len
  182:5   error  Trailing spaces not allowed                                                                    no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\ideal\view\IdealControlPanel.js
  91:1  error  Trailing spaces not allowed  no-trailing-spaces

C:\Users\veillette\phetsims\gas-properties\js\ideal\view\IdealScreenView.js
  60:36  error  Trailing spaces not allowed  no-trailing-spaces

✖ 106 problems (106 errors, 0 warnings)
  32 errors and 0 warnings potentially fixable with the `--fix` option.
Fatal error: 106 Lint Errors
```

**Comment by pixelzoom** [comment]
Thanks @veillette.   Could you please reference #109 in code-review issues that you create, so that they are linked?

**Comment by veillette** [comment]
I edited my comments so that they are linked both ways.

**Comment by pixelzoom** [comment]
The code review item is:

> - [ ] Does linting with "sim_es6_eslintrc_review.js" reveal any problems that should be fixed?

The short answer is no, this doesn't reveal any problems that need to be fixed.  So I'm going to close this issue.

The longer answer is that 100% of these don't need to be fixed, and the majority of them are actually contrary to PhET coding conventions.   So after spending a little time reformatting files in an attempt to remove trailing spaces that WebStorm can't find, I decided that this is busy work that doesn't need to be done.   And I reopened https://github.com/phetsims/gas-properties/issues/114 to revisit why we're doing this extra in the first place.   So it was good that this issue was opened, because it revealed general problems.
